apps/trading/client-pages/market/trade-panels.tsx

```diff
@@ -69,56 +69,83 @@ export const TradePanels = ({ market, pinnedAsset }: TradePanelsProps) => {
-      <div>
-        <MarketBanner market={market} />
-      </div>
-      <div>{renderMenu()}</div>
-      <div className="h-full relative">
-        <AutoSizer>
-          {({ width, height }) => (
-            <div style={{ width, height }} className="overflow-auto">
-              {renderView()}
-            </div>
-          )}
-        </AutoSizer>
-      </div>
-      <div className="flex flex-nowrap overflow-x-auto max-w-full border-t border-default">
-        {Object.keys(TradingViews)
-          // filter to control available views for the current market
-          // eg only perps should get the funding views
-          .filter((_key) => {
-            const key = _key as TradingView;
-            const perpOnlyViews = ['funding', 'fundingPayments'];
-
-            if (
-              market?.tradableInstrument.instrument.product.__typename ===
-              'Perpetual'
-            ) {
+    <div className="h-full flex flex-col lg:grid grid-rows-[min-content_min-content_1fr_min-content]">
+      <div className="flex flex-col w-full overflow-hidden">
+        <div className="flex flex-nowrap overflow-x-auto max-w-full border-t border-default">
+          {['chart', 'orderbook', 'trades', 'liquidity', 'fundingPayments']
+            // filter to control available views for the current market
+            // e.g. only perpetuals should get the funding views
+            .filter((_key) => {
+              const key = _key as TradingView;
+              const perpOnlyViews = ['funding', 'fundingPayments'];
+
+              if (
+                market?.tradableInstrument.instrument.product.__typename ===
+                'Perpetual'
+              ) {
+                return true;
+              }
+
+              if (perpOnlyViews.includes(key)) {
+                return false;
+              }
+
               return true;
-            }
-
-            if (perpOnlyViews.includes(key)) {
-              return false;
-            }
-
-            return true;
-          })
-          .map((_key) => {
-            const key = _key as TradingView;
-            const isActive = view === key;
-            return (
-              <ViewButton
-                key={key}
-                view={key}
-                isActive={isActive}
-                onClick={() => {
-                  setView(key);
-                }}
-              />
-            );
-          })}
+            })
+            .map((_key) => {
+              const key = _key as TradingView;
+              const isActive = view1 === key;
+              return (
+                <ViewButton
+                  key={key}
+                  view={key}
+                  isActive={isActive}
+                  onClick={() => {
+                    setView1(key);
+                  }}
+                />
+              );
+            })}
+        </div>
+        <div className="h-[376px] sm:h-[460px] lg:h-full relative">
```

Comment:
(non blocking): I think we can probably make a better experience by
A) not define heights here and actually use the grid in the wrapping element
B) use percentage or viewport units

Reply:
updated!